WordPress/gutenberg
# Font Library: fix duplicate variants

## What?
This PR adds checks to ensure that multiple file types of the same font variant are not added, so that only one variation per font style and weight is stored. 

## Why?
Fixes #54253

## How?
For fonts where the font family does not yet exist, checks the list of new font faces and compares their font weight and style, and if there is already an existing font face, skips adding the incoming one.

For font faces where the font family exists, adds a new function `get_intersecting_font_faces` which returns an array of font faces to be deleted and replaced by the incoming font face.

## Testing Instructions
1. Simultaneously (batch) upload two or more font faces of the same family, style, and weight, but different file types. 
2. Ensure only the first variant is added / used.
3. Upload a font face to an existing font family with the same style and weight, but different file type.
4. Ensure the uploaded version is used / replaces the existing one, and an additional variant is not created.

### Testing Instructions for Keyboard


## Screenshots or screencast
